example/lib/polymorphism.dart

```diff
@@ -0,0 +1,72 @@
+library polymorphism;
+
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
+
+part 'polymorphism.g.dart';
+
+/// Example of using polymorphism.
+///
+/// Built Value classes and builders can implement interfaces and use mixins as
+/// any other class.
+///
+/// One special case is if you want to `rebuild` instances via an interface.
+/// Then, the interface needs to implement `Built`, but should not itself be
+/// instantiated. You can do this by marking the class `instantiable: false`,
+/// as here.
+///
+/// Very little code is generated for a non-instantiable Built Value; just the
+/// interface for the builder. You can write this yourself if you prefer, then
+/// nothing will be generated.
+@BuiltValue(instantiable: false)
+abstract class Animal extends Object
+    with Walker
+    implements Built<Animal, AnimalBuilder> {
+  @override
+  int get legs;
+}
+
+/// `Cat` implements the non-instantiable Built Value `Animal`. The generated
+/// builder `CatBuilder` automatically extends 'AnimalBuilder`. This allows you
+/// to use `Cat` as an `Animal`.
+abstract class Cat extends Object
+    with Walker
+    implements Animal, Built<Cat, CatBuilder> {
+  /// Serializer field makes the built_value serializable.
+  static Serializer<Cat> get serializer => _$catSerializer;
+
+  bool get tail;
+
+  factory Cat([updates(CatBuilder b)]) = _$Cat;
+  Cat._();
+}
+
+/// `Fish` implements the non-instantiable Built Value `Animal`. The generated
+/// builder `FishBuilder` automatically extends 'AnimalBuilder`. This allows
+/// you to use `Fish` as an `Animal`.
+abstract class Fish extends Object
+    with Swimmer, Walker
+    implements Animal, Built<Fish, FishBuilder> {
+  /// Serializer field makes the built_value serializable.
+  static Serializer<Fish> get serializer => _$fishSerializer;
+
+  @override
+  int get fins;
+
+  factory Fish([updates(FishBuilder b)]) = _$Fish;
+  Fish._();
+}
+
+/// As with any Built Value, behaviour can be added via mixins.
+abstract class Walker {
+  int get legs;
+
+  bool get canWalk => legs > 0;
```

Comment:
[flame war:] One-legged walker doesn't walk, it _hops_. Only 2+ legs can walk.

Reply:
@filiph @davidmorgan  How is it possible to force a class using a mixin to override a getter provided by this mixin? Something similar to how `AutomaticKeepAliveClientMixin` forces you to override `wantKeepAlive`.

Reply:
Not sure if I remember the question correctly, but here goes:

```dart
mixin AutomaticKeepAliveClientMixin {
  void foo() {
    // actually do something useful here
  }

  bool get wantKeepAlive;
}
```

Now `wantKeepAlive` is abstract and when you mix in `AutomaticKeepAliveClientMixin` into a non-abstract class, you should be forced to implement `wantKeepAlive`.

Reply:
@filiph I was too tired to notice the class I was mix in my mixin was abstract, so of course it's not going to yell for an override. Thought there was more to it, but it's pretty straight forward. Thanks!